### ISSUE TITLE
Discord Support

### ISSUE THREAD

**Issue by MoonBarc** [keel-hq/keel#551]
You have hipchat and slack, it would make sense for a discord integration as well. They have a really good API and I feel like it would be a cool addition to Keel.

**Comment by Gabisonfire** [comment]
In the meanwhile I made a simple bridge as a workaround: https://hub.docker.com/repository/docker/gabisonfire/discord-keel-bridge

**Comment by unixfox** [comment]
@Gabisonfire I guess this only works for keel notifications and not keel approvals?

**Comment by MoonBarc** [comment]
Keel approvals would be impossible with discord webhooks unfortunately. I believe it would need to be a "bot" to listen to messages and reactions.

**Comment by unixfox** [comment]
You are right, it needs an active bot and this bot needs to fetch the approvals from keel api (see https://keel.sh/docs/#managing-approvals-through-http-endpoint).

**Comment by Gabisonfire** [comment]
@unixfox Yeah this is only one way. Although coding a discord but is pretty easy, I don't have an incentive to do it at the moment as I'm caught up in other project. Anyways, depending on what the dev team is planning, it would be best to code it in Go.

**Comment by hotpheex** [comment]
Heya, just opened up a PR to add basic Discord notification support.
I had a quick look into how much effort an approval bot would be and it seems like unlike Slack, the discord bot would have to expose an endpoint to the internet for Discord servers to call back directly. Seems possible but a bit of a pain so just sticking to simple notifications for now.

https://github.com/keel-hq/keel/pull/735
